lib/fluent/plugin/parser_multiline.rb

```diff
@@ -24,9 +24,32 @@ class MultilineParser < Parser
 
       desc 'Specify regexp pattern for start line of multiple lines'
       config_param :format_firstline, :string, default: nil
+      desc 'Enable an option returning line as unmatched_line'
+      config_param :unmatched_lines, :string, default: nil
 
       FORMAT_MAX_NUM = 20
 
+      class MultilineRegexpParser < Fluent::Plugin::RegexpParser
+        def parse(text)
+          m = @expression.match(text)
+          unless m
+            yield nil, nil
+            return m
+          end
+
+          r = {}
+          m.names.each do |name|
+            if (value = m[name])
+              r[name] = value
+            end
+          end
+
+          time, record = convert_values(parse_time(r), r)
+
+          yield(time, record)
+          m
+        end
+      end
       def configure(conf)
```

Comment:
putting empty line between `end` and `def` is better.

Reply:
fixed https://github.com/fluent/fluentd/pull/2837/commits/237284a99707075603b9d46d6fcd36b963c99b8a